### DISCUSSION TITLE
Clip and snapshot storage

### DISCUSSION THREAD

**Discussion by picotrain77** [blakeblackshear/frigate#1585] [Ask A Question]
Hi guys
Can someone explain a little about the frigate storage principles. I am running frigate in Hassio on a NUC and have the HA addon all set up and running.

I have tried setting different retentions on all snapshots and clips but not matter what I set them too nothing appears to get deleted, well visually it does not anyway.

In the HA media folders I still have every single clip and snapshot from the day I installed it.

I would not mind so much as I have plenty of space free on the NUC however it can't continue indefinitely and it's making my automated HA snapshots fail.

Just a little understanding of the storage and retention would be great so I can understand a little better. Many thanks

**Comment by blakeblackshear** [comment]
It should be deleting things past the retention period. You should open an issue and provide the requested info in the template.

**Reply by picotrain77** [reply]
> It should be deleting things past the retention period.

Thanks for the reply, I have had a little dig around and can now confirm that the clips and snapshots from my current cameras in the config are correctly deleting as per the retention period.

The reason I noted an issue is that when first setting up the config yml I used test cameras etc, once happy with the setup and almost completely rewrote the config with all the correct cameras and naming conventions.

However for some reason no amount of resetting, rebuilding and reloading will get rid of the the entries relating to the original test cameras, I can still see all the home assistant entities passed to the add-on (having to currently disable them) but cant get rid of them and similarly in media folder have folders populated and named relating to those original test cameras that don't even exist.

That's where I was seeing the additional storage issues, so just need to work out how to flush these out of the system?

Ps love your work 👍🏻

**Reply by CoxieAU** [reply]
I had the same issue and found that deleting the MQTT folders for the test cameras plus a reboot of hassio fixed it

**Reply by picotrain77** [reply]
> I had the same issue and found that deleting the MQTT folders for the test cameras plus a reboot of hassio fixed it 

I assume you mean the media folders here?

**Reply by CoxieAU** [reply]
no on the MQTT server itself, i used an MQTT client to remove the topics
